dashboard/app/helpers/application_helper.rb

```diff
@@ -117,11 +117,7 @@ def teacher_dashboard_url
   end
 
   def teacher_dashboard_section_progress_url(section)
-    CDO.code_org_url "/teacher-dashboard#/sections/#{section.id}"
-  end
-
-  def teacher_dashboard_student_progress_url(section, user)
-    CDO.code_org_url "/teacher-dashboard#/sections/#{section.id}/student/#{user.id}"
+    "/teacher_dashboard/sections/#{section.id}/progress"
```

Comment:
this diff looks weird, so clarifying below:

1. updating `teacher_dashboard_section_progress_url` method to return "/teacher_dashboard/sections/#{section.id}/progress"
2. removing `teacher_dashboard_student_progress_url` as it was only used in [_teacher.html.haml](https://github.com/code-dot-org/code-dot-org/blob/939d289331e173918793d18587577131b930c738/dashboard/app/views/levels/_teacher.html.haml#L18), which is being removed (and is explained in more detail by [this comment](https://github.com/code-dot-org/code-dot-org/pull/27868/files/9af67e8281397d9d1ea03f1b592cde6d0d3ed329#r271966650))